controller/volume_controller.go

```diff
@@ -1493,6 +1499,9 @@ func (vc *VolumeController) ReconcileVolumeState(v *longhorn.Volume, es map[stri
 			if r.Spec.NodeID == "" {
 				continue
 			}
+			if vc.isVolumeUpgrading(v) && r.Status.InstanceStatus.CurrentState != longhorn.InstanceStateRunning {
+				continue
+			}
```

Comment:
Shouldn't it check if the instance-manager-r is starting or running instead?

Reply:
Modified, but I think `InstanceStatus.CurrentState` only `running` might be safer to schedule them to engine `replicaAddressMap`.

Reply:
What I mean is, the below `canInstanceManagerLaunchReplica` should return false if the related instance-manager-r is not running or starting.

Reply:
Got it and Modified. Wait for IM-r is starting when `canInstanceManagerLaunchReplica` returns `false`.